datafusion/src/physical_plan/planner.rs

```diff
@@ -210,6 +243,24 @@ impl PhysicalPlanner for DefaultPhysicalPlanner {
         let plan = self.create_initial_plan(logical_plan, ctx_state)?;
         self.optimize_plan(plan, ctx_state)
     }
+
+    /// Create a physical expression from a logical expression
+    /// suitable for evaluation
+    ///
+    /// `e`: the expression to convert
+    ///
+    /// `input_dfschema`: the logical plan schema for evaluating `e`
+    ///
+    /// `input_schema`: the physical schema for evaluating `e`
+    fn create_physical_expr(
+        &self,
+        e: &Expr,
+        input_dfschema: &DFSchema,
+        input_schema: &Schema,
+        ctx_state: &ExecutionContextState,
+    ) -> Result<Arc<dyn PhysicalExpr>> {
+        self.create_physical_expr(e, input_dfschema, input_schema, ctx_state)
```

Comment:
From here it looks like infinite recursion, but I might be wrong?

Reply:
It somehow knows to call `DefaultPhysicalPlanner::create_physical_expr`  -- I will make this clear